base/linalg/lapack.jl

```diff
@@ -1700,8 +1700,8 @@ for (f, elty) in ((:dggsvd3_, :Float64),
                 throw(DimensionMismatch("B has second dimension $(size(B,2)) but needs $n"))
             end
             p = size(B, 1)
-            k = Ref{BlasInt}()
-            l = Ref{BlasInt}()
+            k = Array(BlasInt, 1)
+            l = Array(BlasInt, 1)
```

Comment:
Strictly speaking I guess `k` and `l` were okay using `Ref` since none of the macros were referring to them, and it might look a little more natural below to use `k[1]` instead of `k[]` now if it's going to be an `Array`, but for now it's the same.